bodo/submit/spawner.py

```diff
@@ -556,6 +556,7 @@ def reset(self):
             # We might not be able to log during process teardown
             pass
         self.worker_intercomm.bcast(CommandType.EXIT.value, root=self.bcast_root)
+        self.worker_intercomm.Disconnect()
```

Comment:
`Disconnect` seems necessary for proper clean up.
https://mpi4py.readthedocs.io/en/stable/tutorial.html#dynamic-process-management